staging/src/k8s.io/apimachinery/pkg/util/errors/errors_test.go

```diff
@@ -430,3 +431,97 @@ func TestAggregateGoroutines(t *testing.T) {
 		}
 	}
 }
+
+type alwaysMatchingError struct{}
+
+func (_ alwaysMatchingError) Error() string {
+	return "error"
+}
+
+func (_ alwaysMatchingError) Is(_ error) bool {
+	return true
+}
+
+type someError struct{}
+
+func (_ someError) Error() string {
+	return "err"
+}
+
+func TestAggregateWithErrorsIs(t *testing.T) {
+	testCases := []struct {
+		name         string
+		aggregate    aggregate
+		matchAgainst error
+		expectMatch  bool
+	}{
+		{
+			name:         "no match",
+			aggregate:    aggregate{errors.New("my-error"), errors.New("my-other-error")},
+			matchAgainst: fmt.Errorf("no entry %s", "here"),
+		},
+		{
+			name:         "match via .Is()",
+			aggregate:    aggregate{errors.New("forbidden"), alwaysMatchingError{}},
+			matchAgainst: errors.New("unauthorized"),
+			expectMatch:  true,
+		},
+		{
+			name:         "match via equality",
+			aggregate:    aggregate{errors.New("err"), someError{}},
+			matchAgainst: someError{},
+			expectMatch:  true,
+		},
+		{
+			name:         "match via nested aggregate",
+			aggregate:    aggregate{errors.New("closed today"), aggregate{aggregate{someError{}}}},
+			matchAgainst: someError{},
+			expectMatch:  true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			result := errors.Is(tc.aggregate, tc.matchAgainst)
+			if result != tc.expectMatch {
+				t.Errorf("expected match: %t, got match: %t", tc.expectMatch, result)
+			}
+		})
+	}
+}
+
+func TestAggregateWithErrorsAs(t *testing.T) {
+	testCases := []struct {
+		name         string
+		aggregate    aggregate
+		matchAgainst interface{}
+		expectMatch  bool
+	}{
+		{
+			name:         "no match",
+			aggregate:    aggregate{errors.New("my-error"), errors.New("my-other-error")},
+			matchAgainst: &someError{},
+		},
+		{
+			name:         "match",
+			aggregate:    aggregate{errors.New("forbidden"), someError{}},
+			matchAgainst: &someError{},
+			expectMatch:  true,
+		},
+		{
+			name:         "match via nested aggregate",
+			aggregate:    aggregate{errors.New("closed today"), aggregate{aggregate{someError{}}}},
+			matchAgainst: &someError{},
+			expectMatch:  true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			result := errors.As(tc.aggregate, tc.matchAgainst)
```

Comment:
nit: check that `target` is actually assigned (applies to both test files).

Reply:
Oh and for both the `Is`/`As` tests I think you want a test to confirm that the implementation short-circuits on the first matching error.

Reply:
Oh and for both the `Is`/`As` tests I think you want a test to confirm that the implementation short-circuits on the first matching error.

Reply:
Done